src/core/service.c

```diff
@@ -2290,6 +2290,19 @@ static int service_deserialize_item(Unit *u, const char *key, const char *value,
                 else {
                         asynchronous_close(s->socket_fd);
                         s->socket_fd = fdset_remove(fds, fd);
+                        if (UNIT_ISSET(s->accept_socket)) {
+                                Socket* socket = SOCKET(UNIT_DEREF(s->accept_socket));
+
+                                if (socket->max_connections_per_source > 0) {
+                                        SocketPeer *peer;
+
+                                        /* Make a best-effort attempt at bumping the connection count */
+                                        if (socket_acquire_peer(socket, s->socket_fd, &peer) > 0) {
+                                                socket_peer_unref(s->peer);
+                                                s->peer = peer;
+                                        }
```

Comment:
hmm, this will only work if the accept_socket is serialized/deserialized before the socket-fd. I think this at least deserves a comment.

That said, I think it would be better to move the above into service_coldplug(), which is where we reconstruct the runtime objects off the serialized state again, and this is kinda appropriate for that.